### PR TITLE
RavenDB-18545 - Fix - Can't select a field that its name is equal to …

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18545

### Additional description

Fix - Can't select a field that its name is equal to some alias in Query.

### Type of change

- Bug fix

### How risky is the change?

- Low

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/Session/Tokens/GroupByKeyToken.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public override void WriteTo(StringBuilder writer)
 
             writer
                 .Append(" as ")
-                .Append(_projectedName);
+                .Append($"\'{_projectedName}\'");
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/Tokens/GroupByKeyToken.cs]: What is the name also contains `'`?

Comment (ppekrol) [reply src/Raven.Client/Documents/Session/Tokens/GroupByKeyToken.cs]: I do not think we are doing that anywhere.

Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Tokens/GroupByKeyToken.cs]: It should be:
```
writer
    .Append(" as '")
    .Append(_projectedName)
    .Append("'");   
```

You will avoid string allocation.

File: test/SlowTests/Issues/RavenDB-18545.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                                        .Select(x => x.Group).ToList();
+
+                    Assert.NotEmpty(l);
+                    Assert.Equal(1, l.Count);
+                    Assert.Equal(l[0], job.Group);
+                }
+            }
+        }
+
+        class Job
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB-18545.cs]: private

Comment (shaharhikri) [reply test/SlowTests/Issues/RavenDB-18545.cs]: Done ( I'm pretty sure it's private by default)

File: src/Raven.Client/Documents/Session/Tokens/GroupByKeyToken.cs (hunk 1 above)
Comment (shaharhikri) [reply src/Raven.Client/Documents/Session/Tokens/GroupByKeyToken.cs]: Done

**Pull request merged**